### DISCUSSION TITLE
UnicodeDecodeError when reading config file - it has worked with an oder version of borg...

### DISCUSSION THREAD

**Discussion by sophonet** [borgbackup/borg#7551] [Q&A]
Hi,

I am using borg backup from a container(with mounted host directories) and have done this successfully in the last two years. Recently I have upgraded my system and have switched to a more recent version I guess (the Dockerfile does not contain version numbers though, so I cannot easily tell which version I have used before). Now I cannot access the repository anymore due to a UnicodeDecodeError on the config file, see below. Indeed, the repository's config file is full of binary characters and not readable (starts with ��B�F��) - and according to the documentation, its structure should rather be a text file in .ini format? I have tried setting the LANG variable according to the documentation to various entries, but without success.

I have created the repository with the '--encryption=repokey-blake2' flag.

Is there anything I need to consider in order to get borg running again or do I need to recreate the repository from scratch? I am a bit worried that this situation might happen when I need to restore a backup, so I would like to understand the reason behind it.

Thanks and best regards,

sophonet

  File "/usr/lib/python3.11/site-packages/borg/repository.py", line 438, in open
    self.config.read_file(fd)
  File "/usr/lib/python3.11/configparser.py", line 734, in read_file
    self._read(f, source)
  File "/usr/lib/python3.11/configparser.py", line 1036, in _read
    for lineno, line in enumerate(fp, start=1):
  File "<frozen codecs>", line 322, in decode
UnicodeDecodeError: 'utf-8' codec can't decode byte 0x82 in position 1: invalid start byte

**Comment by ThomasWaldmann** [comment]
That sounds bad. Seems like the filesystem or storage (where the repo is on) does not work correctly in your setup.

If you made the borg key backup as `borg init` advised you to do, you could create a new config using it and regain access to your borg repo. If you didn't and you lost the key in the repo config, you can't decrypt the data in the repo any more.

Please do a `ls -l repo_directory` and post the output.

**Comment by sophonet** [comment]
Thanks for picking this up!  This is the output of ls -l in the repo_directory.

-r--------   1 root root   155 Dec 18  2021 config
drwx------ 258 root root  4096 Dec 18  2021 data
drwx------   2 root root 45056 Jan 11 09:07 index
drwx------   2 root root  4096 Dec 18  2021 keys
drwx------   2 root root  4096 Jan 11 09:07 locks
drwx------   2 root root 20480 Jan 11 09:07 snapshots

Again, incremental backups have worked in the past and I just don't know why they don't do it anymore and would like to understand.

**Comment by ThomasWaldmann** [comment]
that does not quite look like a `borg` repo. `restic` maybe?

**Comment by sophonet** [comment]
Apologies, @ThomasWaldmann you are right, I have picked an old backup medium from an earlier obsolete restic backup (I am not using restic anymore). Again, thanks for the clarification and support!

**Comment by ThomasWaldmann** [comment]
BTW, borg has a README inside the repo directory which states that that is a borg repo.
